mycity/mycity/utilities/arcgis_utils.py

```diff
@@ -0,0 +1,283 @@
+    
+    # (x, y) coordinates of origin address
+    incidents = "{},{}".format(origin_address['x'], origin_address['y'])
+
+    # List of (x, y) coordinates of possible destinations
+    facility_list = []
+    facility_key_list = []
+    for k in destination_addresses.keys():
+        if k[0] != "" and k[1] != "":
+            temp_str = "{},{}".format(k[0], k[1])
+            facility_list.append(temp_str)
+            facility_key_list.append(k)
+    
+    # Separate destination coordinates with ";"
+    facilities = ";".join(facility_list)
+    params = {
+            'f': 'json',
+            'token': api_access_token,
+            'returnDirections': 'false',
+            'returnCFRoutes': 'true',
+            'incidents': incidents,
+            'facilities': facilities
+            }
+
+    formatted_tuple = format_multipart_form_request(base_url, params)
+    body_as_string = formatted_tuple[0]
+    updated_header = formatted_tuple[1]
+    # POST request over network
+    response = _post_request(base_url, body_as_string, updated_header)
+
+    if response.status_code != 200:
+        logger.debug("Response Error: {}".format(str(response.status_code)))
+        return None
+    else:
+        response_json = response.json()
+        logger.debug("Response JSON: {}".format(str(response_json)))
+
+        routes = response_json['routes']
+        features = routes['features']
+        attributes = features[0]['attributes']
+        facility_id = attributes['FacilityID']
+        travel_time_in_minutes = attributes['Total_TravelTime']
+        travel_distance_in_miles = attributes['Total_Miles']
+
+        formatted_travel_time_in_minutes = _format_float(float(travel_time_in_minutes))
+        formatted_travel_distance_in_miles = _format_float(float(travel_distance_in_miles))
+
+        travel_time_string = "{} minutes".format(formatted_travel_time_in_minutes)
+        travel_distance_string = "{} miles".format(formatted_travel_distance_in_miles)
+
+        facility_key_index = int(facility_id) - 1
+        facility_key = facility_key_list[facility_key_index]
+        facility_address = destination_addresses[facility_key]
+
+        destination_dict = {
+                'Address': facility_address,
+                'Driving_time': travel_time_string,
+                'Driving_distance': travel_distance_string
+                }
+
+        logger.debug("Returning closest destination: {}".format(str(destination_dict)))
+        return destination_dict
+
+
+def format_multipart_form_request(url, params):
+    """
+    Formats a multipart/form POST request
+    for requests module properly for
+    ESRI ARCGis API
+
+    :param url: String containg URL of API
+    :param params: Dictionary of paramters
+    :return Two-Tuple containing 1) String representing
+        params to be passed as data to POST request
+        and 2) Dictionary with modified headers
+    """
+    logger.debug("URL: {}, Params: {}".format(url, str(params)))
```

Comment:
I think this is fine. Debug level logs will be ignored when the app is live and we've set the threshold to `warning`. This format looks good to me as it logs useful values -- the thing we were doing before was logging the entire request object in tons of places, rather than the specific values that are relevant. We still have that in a lot of places. Would be a good starter ticket.